ownership-chain/pallets/block-rewards-source/src/lib.rs

```diff
@@ -1,26 +1,47 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use frame_support::pallet_prelude::*;
+// External crates
+use frame_support::{
+	pallet_prelude::*,
+	traits::{Currency, ExistenceRequirement, LockableCurrency, ReservableCurrency},
+};
 
-// Crate's internal imports
+// Standard library
+use sp_runtime::ArithmeticError;
+
+// Your own modules
+mod benchmarking;
+pub mod weights;
+
+// Re-exports
 pub use pallet::*;
+pub use weights::WeightInfo;
 
 #[frame_support::pallet]
 pub mod pallet {
+
 	use super::*;
 
 	/// The current storage version.
 	const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
 
+	pub type BalanceOf<T> =
+		<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
 	#[pallet::pallet]
 	#[pallet::storage_version(STORAGE_VERSION)]
 	pub struct Pallet<T>(_);
 
 	/// Configure the pallet by specifying the parameters and types on which it depends.
 	#[pallet::config]
 	pub trait Config: frame_system::Config {
-		/// Because this pallet emits events, it depends on the runtime's definition of an event.
-		type RuntimeEvent: From<Event<Self>> + IsType<<Self as frame_system::Config>::RuntimeEvent>;
+		/// Weight information for extrinsics in this pallet.
+		type WeightInfo: WeightInfo;
+
+		/// The currency type
+		type Currency: Currency<Self::AccountId>
+			+ ReservableCurrency<Self::AccountId>
```

Comment:
you don't need these, only `Currency` is needed. Even better if you use `Mutate` since `Currency` will be deprecated soon

Reply:
good point, but for now I'll leave just as `Currency` since use `Mutate` requires more changes in the code because `Mutate` trait functions differs

Reply:
for functions that we use in this pallet, interface is the same. for `staking`, yes it might differ for freeze/locking. 

the comment here was about removing unnecessary bounds, replacing with `Mutate` was kind of bonus

Reply:
you are right, here we only use transfer and it doesn't affect us

Reply:
I found that `Mutate` uses `Unbalance` type instead of `Balance`, I'll leave as it is so far and address it in future implementations